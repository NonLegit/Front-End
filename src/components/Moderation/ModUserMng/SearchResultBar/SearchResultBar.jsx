// import React from 'react';
import { Typography, Box } from '@mui/material';
import { StyledSearchResultBar } from './styles';
import { UserMngButton } from '../styles';

/**
 * the whole page of no result that appears when try to seach on a user in user managment section in moderatio page
 * @component
 * @property {string} resultNumber the number of results from the query after search
 * @property {object} filteredData the result data after searching
 * @property {function} childToParent a function to pass result data from child to parent
 * @return {React.Component} - no result page
 */

function SearchResultBar(props) {
  const {
    resultNumber, filteredData, childToParent,
  } = props;
  return (
    <StyledSearchResultBar>
      <Box display="flex">
        <Typography
          fontSize="14px"
          fontWeight={600}
          color="#1C1C1C"
          paddingLeft="15px"
        >
          {resultNumber}
          {' '}
          search result for&nbsp;
        </Typography>
        <Box display="flex">
          {
              filteredData.map((users) => {
                const { user } = users;
                return (
                  <Typography
                    color="#0079D3"
                    fontSize="14px"
                    fontWeight={500}
                  >
                    {`'${user.userName}'`}
                    &nbsp;
                  </Typography>
                );
              })
      }
        </Box>
      </Box>
      <UserMngButton onClick={() => childToParent('')}>
        See all
      </UserMngButton>
    </StyledSearchResultBar>
  );
}

export default SearchResultBar;
