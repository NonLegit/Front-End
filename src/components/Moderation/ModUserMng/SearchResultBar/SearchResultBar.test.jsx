import renderer from 'react-test-renderer';
import SearchResultBar from './SearchResultBar';

const data = () => {};

test('test snapshot', async () => {
  const tree = renderer.create(
    <SearchResultBar
      resultNumber="350k"
      filteredData={[]}
      childToParent={data}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
