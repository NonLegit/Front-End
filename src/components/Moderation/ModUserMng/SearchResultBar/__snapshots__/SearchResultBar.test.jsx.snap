// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<div
  className="MuiBox-root css-1yejm7w"
>
  <div
    className="MuiBox-root css-k008qs"
  >
    <p
      className="MuiTypography-root MuiTypography-body1 css-10bhfgi-MuiTypography-root"
    >
      350k
       
      search result forÂ 
    </p>
    <div
      className="MuiBox-root css-k008qs"
    />
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium css-p5wwkm-MuiButtonBase-root-MuiButton-root"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onContextMenu={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    See all
  </button>
</div>
`;
